### PR TITLE
feat: update record classes for CoMLOPs format

### PR DESCRIPTION
**Pull request by ktro2828**
## Description

This PR includes following changes:
1. Update fields of `ego_pose` and store data recorded by INS
2. Add support of generating of `vehicle_state.json`

### Configuration updates

As described in `config/convert_comlops_rosbag2_to_non_annotated_t4_sample.yaml`, please edit fields in `conversion` as follows:

```yaml
conversion:
  ...
  with_ins: true            # whether to use INS messages as a ego state instead of `/tf`
  with_vehicle_status: true # whether to generate `vehicle_state.json`
```

### EgoPose

- If `with_ins=True` the converter initializes `INSHandler` and load ego poses using `odometry`, `imu` and `nav_sat_fix` instead of `/tf` message.
  - `INSHandler` retrieves an ego state interpolated by linear at the corresponding timestamp.

The sample output of `ego_pose.json` is as below:

```json
    {
        "token": "082008187ebfca8a25c9ce8a500fb2c3",
        "translation": [
            -6.2490851223989035,
            -3.0805003909970616,
            -2.4697036630417264
        ],
        "rotation": [
            4.329780281177467e-17,
            0.7071067811865476,
            0.7071067811865475,
            4.329780281177466e-17
        ],
        "timestamp": 1724306789117423,
        "twist": [
            0.0,
            0.0,
            0.0,
            0.0,
            0.0,
            0.0
        ],
        "acceleration": [
            0.11715737026959112,
            -0.9582391259005323,
            -9.723479226352445
        ],
        "geocoordinate": [
            35.624468375999996,
            139.74287536399999,
            61.35459899902344
        ]
    },
```

### VechileState

- If `with_vechile_status=True`, the converter initializes `VehicleStatusHandler` and load certain vehicle states.
  - Note that, currently, any records in `vehicle_state.json` are **not associated with the other records** such as `sample.json`.

The sample output of `vehicle_state.json` is as below:

```json
    {
        "token": "269572c280bd5cf9630ca542e6a60185",
        "timestamp": 1724306784277396,
        "accel_pedal": 0.0,
        "brake_pedal": 1.0,
        "steer_pedal": 0.6063521901837905,
        "steering_tire_angle": 0.6063522100448608,
        "steering_wheel_angle": 9.291000366210938,
        "shift_state": "PARK",
        "indicators": {
            "left": "off",
            "right": "on",
            "hazard": "off"
        },
        "additional_info": {
            "speed": 0.0
        }
    },
```

### test command

1. Setting up the environment

```shell
# Added `ros-${ROS_DISTRO}-tf-transformations`
sudo apt install -y ros-${ROS_DISTRO}-sensor-msgs-py ros-${ROS_DISTRO}-rosbag2-storage-mcap ros-${ROS_DISTRO}-radar-msgs ros-${ROS_DISTRO}-tf-transformations

mkdir src -p && vcs import src < build_depends.repos

# Added `oxts_msgs` and `vehicle_msgs`
colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release --packages-up-to autoware_auto_perception_msgs autoware_perception_msgs tier4_perception_msgs oxts_msgs vehicle_msgs
source ./install/setup.bash
```

2. After editing `input_base` in the `config/convert_comlops_rosbag2_to_non_annotated_t4_sample.yaml`, run following command:

```bash
$ python3 -m perception_dataset.convert --config config/convert_comlops_rosbag2_to_non_annotated_t4_sample.yaml
```

### CODE REVIEW

File: perception_dataset/ros2/oxts_msgs/ins_handler.py [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+            header.frame_id = "map"
+            header.stamp = unix_timestamp_to_stamp(timestamp)
+
+            twist_msg = Twist()
+            twist_msg.linear.x = twist[0]
+            twist_msg.linear.y = twist[1]
+            twist_msg.linear.z = twist[2]
+            twist_msg.angular.x = twist[0]
+            twist_msg.angular.y = twist[1]
+            twist_msg.angular.z = twist[2]
```
Comment (miursh) [inline perception_dataset/ros2/oxts_msgs/ins_handler.py:269]:
```suggestion
            twist_msg.angular.x = twist[3]
            twist_msg.angular.y = twist[4]
            twist_msg.angular.z = twist[5]
```

**Comment by ktro2828** [comment]
TODO List

- [ ] Convert the unit of angles in INS from degree to rad
- [ ] Resolve the coordinates origin of twists and accelerations, w.r.t base_link or map
- [ ] Check the accuracy of twists and accelerations

**Comment by miursh** [comment]
@ktro2828 
About the INS unit, I will update the driver code. so there is no need to support unit conversion here.

**Comment by ktro2828** [comment]
@miursh 
Here is my script to check data: [TIER IV INTERNAL LINK](https://tier4.atlassian.net/wiki/spaces/~308143412/pages/3364324420/WIP+rosbag+-+dataset+converter?focusedCommentId=3424453411)

**Review by miursh — APPROVED** [review]
At least the INS part looks good to me.
I think it's okay to merge if we can fix the CI/tests.
(I haven't tested the vehicle_state part, though...)

**Comment by ktro2828** [comment]
I confirmed all CI has been passed at the latest commit: https://github.com/tier4/tier4_perception_dataset/pull/163/commits/c208fe9a75ad0551a0367db6fb00a797d068b0d7.
I'll merge this PR.

**Pull request merged**